azurerm/helpers/azure/app_service.go

```diff
@@ -452,6 +647,354 @@ func FlattenAppServiceCorsSettings(input *web.CorsSettings) []interface{} {
+		facebookSettings := v.([]interface{})
+
+		for _, setting := range facebookSettings {
+			facebookSetting := setting.(map[string]interface{})
+
+			if v, ok := facebookSetting["app_id"]; ok {
+				siteAuthSettingsProperties.FacebookAppID = utils.String(v.(string))
+			}
+
+			if v, ok := facebookSetting["app_secret"]; ok {
+				siteAuthSettingsProperties.FacebookAppSecret = utils.String(v.(string))
+			}
+
+			if v, ok := facebookSetting["oauth_scopes"]; ok {
+				input := v.([]interface{})
+
+				oauthScopes := make([]string, 0)
+				for _, param := range input {
+					oauthScopes = append(oauthScopes, param.(string))
+				}
+
+				siteAuthSettingsProperties.FacebookOAuthScopes = &oauthScopes
+			}
+		}
+	}
+
+	if v, ok := setting["google"]; ok {
+		googleSettings := v.([]interface{})
+
+		for _, setting := range googleSettings {
+			googleSetting := setting.(map[string]interface{})
+
+			if v, ok := googleSetting["client_id"]; ok {
+				siteAuthSettingsProperties.GoogleClientID = utils.String(v.(string))
+			}
+
+			if v, ok := googleSetting["client_secret"]; ok {
+				siteAuthSettingsProperties.GoogleClientSecret = utils.String(v.(string))
+			}
+
+			if v, ok := googleSetting["oauth_scopes"]; ok {
+				input := v.([]interface{})
+
+				oauthScopes := make([]string, 0)
+				for _, param := range input {
+					oauthScopes = append(oauthScopes, param.(string))
+				}
+
+				siteAuthSettingsProperties.GoogleOAuthScopes = &oauthScopes
+			}
+		}
+	}
+
+	if v, ok := setting["microsoft"]; ok {
+		microsoftSettings := v.([]interface{})
+
+		for _, setting := range microsoftSettings {
+			microsoftSetting := setting.(map[string]interface{})
+
+			if v, ok := microsoftSetting["client_id"]; ok {
+				siteAuthSettingsProperties.MicrosoftAccountClientID = utils.String(v.(string))
+			}
+
+			if v, ok := microsoftSetting["client_secret"]; ok {
+				siteAuthSettingsProperties.MicrosoftAccountClientSecret = utils.String(v.(string))
+			}
+
+			if v, ok := microsoftSetting["oauth_scopes"]; ok {
+				input := v.([]interface{})
+
+				oauthScopes := make([]string, 0)
+				for _, param := range input {
+					oauthScopes = append(oauthScopes, param.(string))
+				}
+
+				siteAuthSettingsProperties.MicrosoftAccountOAuthScopes = &oauthScopes
+			}
+		}
+	}
+
+	if v, ok := setting["twitter"]; ok {
+		twitterSettings := v.([]interface{})
+
+		for _, setting := range twitterSettings {
+			twitterSetting := setting.(map[string]interface{})
+
+			if v, ok := twitterSetting["consumer_key"]; ok {
+				siteAuthSettingsProperties.TwitterConsumerKey = utils.String(v.(string))
+			}
+
+			if v, ok := twitterSetting["consumer_secret"]; ok {
+				siteAuthSettingsProperties.TwitterConsumerSecret = utils.String(v.(string))
+			}
+		}
+	}
+
+	return siteAuthSettingsProperties
+}
+
+func FlattenAdditionalLoginParams(input *[]string) map[string]interface{} {
+	result := make(map[string]interface{}, len(*input))
+
+	for _, k := range *input {
+		parts := strings.Split(k, "=")
```

Comment:
there's a potential crash here; could we ensure we've got 2 parts here?

```suggestion
		parts := strings.Split(k, "=")
		if len(parts) != 2 {
		  ...
		}
```